### ISSUE TITLE
Namespace constants being inlined in production build

### ISSUE THREAD

**Issue by aweary** [facebook/react#11920]
I was poking through the UMD `react-dom.production.min.js` build and noticed that the namespaces exported from [`DOMNamespaces`](https://github.com/facebook/react/blob/master/packages/react-dom/src/shared/DOMNamespaces.js) are being inlined as string literals.

[First Example](https://gist.github.com/aweary/da7ed93a94b7c8ddf65c569be61e8356#file-react-dom-min-js-L1779-L1794)
[Second Example](https://gist.github.com/aweary/da7ed93a94b7c8ddf65c569be61e8356#file-react-dom-min-js-L1872-L1873)

This isn't great since they are sort of long strings. I'm not sure if its a bug or expected behavior for some component of the build system, but I thought I'd report it in case something can be configured?

**Comment by gaearon** [comment]
This is intentional:

https://github.com/google/closure-compiler/wiki/FAQ#closure-compiler-inlined-all-my-strings-which-made-my-code-size-bigger-why-did-it-do-that

> The gzip algorithm works by trying to alias sequences of bytes in an optimal way. Aliasing strings manually almost always makes the compressed code size bigger, because it subverts gzip's own algorithm for aliasing. So Closure Compiler will (almost) always inline your strings when it can, because that will make your compressed code smaller.

**Comment by aweary** [comment]
@gaearon interestingly enough, I manually removed each inlined namespace string and aliased it. The resulting file size was 30755 bytes, compared to the original 30770 bytes according to `gzip-size`. It's not a huge difference, but I wonder why it's not compressing better?

Out of curiosity I also aliased a few of the inlined strings that appear most often (`"function"`, `"return"` for example) and the got the size down to 30689 bytes.
